orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt

```diff
@@ -127,8 +127,17 @@ class CompleteStageHandler(
               return@withStage
             }
           }
-
           stage.status = status
+          if (status == FAILED_CONTINUE) {
```

Comment:
I'm curious why only FAILED_CONTINUE here.  What happens if we do this for all statuses, or if not that, maybe also TERMINAL?

Reply:
FAILED_CONTINUE seems sufficient to address https://github.com/spinnaker/spinnaker/issues/6615...does the same exception info show up via some other mechanism for e.g. TERMINAL?  If so, where?

Reply:
In case of TERMINAL status,the exact failed task exception itself is registered in the `stagecontext` exception and is shown.But for FAILED_CONTINUE this was not happening and hence had to restrict for FAILED_CONTINUE only.

Reply:
In case of TERMINAL status,the exact failed task exception itself is registered in the stagecontext exception and is shown.But for FAILED_CONTINUE this was not happening and hence had to restrict for FAILED_CONTINUE only.

Reply:
In this code below, right?
```
        } catch (e: Exception) {
          log.error("Failed to construct after stages for ${stage.name} ${stage.id}", e)

          val exceptionDetails = exceptionHandlers.shouldRetry(e, stage.name + ":ConstructAfterStages")
          stage.context["exception"] = exceptionDetails
          stage.status = TERMINAL
          stage.endTime = clock.millis()
        }
```

Reply:
The exception for failed task is tied to `stagecontext ` in `Runtaskhandler.kt` in case of TERMINAL status.The fix proposes to tie the exceptions after the completion of the stage for FAILED_CONTINUE status.